pkg/session/nontransactional.go

```diff
@@ -275,6 +279,14 @@ func runJobs(ctx context.Context, jobs []job, stmt *ast.NonTransactionalDMLStmt,
 	if shardColumnRefer == nil && stmt.ShardColumn.Name.L != model.ExtraHandleName.L {
 		return nil, errors.New("Non-transactional DML, shard column not found")
 	}
+	switch s := stmt.DMLStmt.(type) {
+	case *ast.DeleteStmt:
+		s.IsNontransactionalDML = true
+	case *ast.UpdateStmt:
+		s.IsNontransactionalDML = true
+	case *ast.InsertStmt:
+		s.IsNontransactionalDML = true
```

Comment:
```suggestion
	case *ast.DeleteStmt, *ast.UpdateStmt, *ast.InsertStmt:
		s.IsNontransactionalDML = true
```

Reply:
There will be build error if we merge them together, because they only implement the common interface `ShardableDMLStmt`.

```bash
pkg/session/nontransactional.go:284:5: s.IsNontransactionalDML undefined (type "github.com/pingcap/tidb/pkg/parser/ast".ShardableDMLStmt has no field or method IsNontransactionalDML)
make: *** [Makefile:231: server] Error 1
```

https://github.com/pingcap/tidb/blob/8acfef88c43fbe50df830363ff00d52346096838/pkg/parser/ast/dml.go#L2707-L2717

Reply:
Got it